### ISSUE TITLE
Installation needs to be simpler

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#9401]
Our current installation process is multi-step, involves the command line, etc. We have data showing that this results in failed installations and abandoned installations.

I think we need to have a single-step install.

I suggest:
- [ ] On Mac OS, have a single DMG that you can drag onto your machine (Apps directory? Home directory?) that includes our git repo, cached binaries, and all the dependencies we can legally bundle. This gets you to the same state you have today, and leaves our current upgrade path intact.
- [ ] On Windows, have an installation app that you can download and run that bundles our git repo, cached binaries, and all the dependencies we can legally bundle, and leaves our current upgrade path intact.
- [ ] On Linux, have a bootstrap script that does the same thing into your home directory.
- [ ] We should pin and bundle all our Dart dependencies. https://github.com/flutter/flutter/issues/6767
- [ ] We should try to shed as many dependencies as we can.

**Comment by eseidelGoogle** [comment]
I suspect @mit-mit would agree. :)

**Comment by mit-mit** [comment]
Yes! Shared a doc with some more thoughts.

**Comment by timsneath** [comment]
cc @csells

**Comment by davidfetter** [comment]
Bump!

**Comment by tonychow1997** [comment]
At least, would a SHA256 checksum be provided to ensure that the downloaded zip file is not corrupted?

**Comment by mit-mit** [comment]
cc @gspencergoog for the checksum part

**Comment by gspencergoog** [comment]
Yes, we could add a checksum to the downloadable packages. I filed https://github.com/flutter/flutter/issues/28465 to address this.

**Comment by superciccio** [comment]
Hello everyone, 
is this issue still open ? I'd like work on it, what's the status ?

Thanks

**Comment by bkonyi** [comment]
Simplified installation beyond what we currently provide isn't prioritized or planned, so I'm going to close this for now. We may decided to revisit this in the future.
